lib/marked.js

```diff
@@ -674,18 +674,21 @@ InlineLexer.prototype.output = function(src) {
           url = url.substring(0, url.length - 1);
       }
 
-      src = src.substring(url.length);
       text = escape(url);
       href = text;
 
-      tokens.push({
-        type: 'link',
-        text: text,
-        title: null,
-        href: href,
-        isUrl: true
-      });
-      continue;
+      if (this.options.urlFilter(url)) {
```

Comment:
what happens if condition is `false`? Before, we were removing the url chars from the `src` source string, now we might run the risk of encountering it again?